### ISSUE TITLE
Type names should be allowed to be prefixed with $

### ISSUE THREAD

**Issue by jacehensley-wf** [dart-lang/linter#220]
`camel_case_types` should allow types that start with `$`.

__GOOD__:
```dart
class $SomeClass {

}
```
